File: 프로그래머스/lv1/장정윤/이웃한 칸.py
from collections import deque

def solution(board, h, w):
    answer = 0
    dx = [-1,1,0,0]
    dy = [0,0,-1,1]

    q = deque()
    q.append((h,w))

    while q:
        x,y = q.popleft()

        color = board[x][y]

        for i in range(4):
            nx = x + dx[i]
            ny = y + dy[i]

            if 0<=nx<len(board[0]) and 0<=ny<len(board):
                if board[nx][ny] == board[x][y]:
                    answer += 1
    return answer

File: 프로그래머스/lv1/이재표/이웃한 칸.java
import java.util.*;
class Solution {
    public int solution(String[][] board, int h, int w) {
        int answer = 0;
        int[] dx={0,0,-1,1};
        int[] dy={-1,1,0,0};
        String color=board[h][w];
        for(int i=0;i<4;i++){
            if(0<=h+dy[i]&&h+dy[i]<board.length&&0<=w+dx[i]&&w+dx[i]<board[0].length){
                int y=h+dy[i];
                int x=w+dx[i];
                if(board[y][x].equals(color)){
                    answer++;
                }
            }
        }
        return answer;
    }
}
